ui/StatusQ/include/StatusQ/keychain.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <QObject>
+#include <QFuture>
+
+#ifdef __OBJC__
+#include <LocalAuthentication/LAContext.h>
+#else
+class LAContext;
+#endif
```

Comment:
```suggestion
class LAContext;
#17147 
```

It can be forward declaration for both cases. Include can be moved to `.mm` file.

Reply:
@micieslak nope, because of in `LAContext.h` it's defined in Objective-C:
```objc
/// Class that represents an authentication context.
///
/// @discussion This context can be used for evaluating policies.
///
/// @see LAPolicy
API_AVAILABLE(macos(10.10), ios(8.0), watchos(3.0)) API_UNAVAILABLE(tvos)
@interface LAContext : NSObject
```

And it doesn't compile without this `#ifdef __OBJC__` on macs:
```
In file included from /Users/igorsirotin/Repositories/Status/status-desktop/ui/StatusQ/src/keychain_osx.mm:10:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/System/Library/Frameworks/LocalAuthentication.framework/Headers/LocalAuthentication.h:8:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.2.sdk/System/Library/Frameworks/LocalAuthentication.framework/Headers/LAContext.h:98:12: error: redefinition of 'LAContext' as different kind of symbol
   98 | @interface LAContext : NSObject
      |            ^
/Users/igorsirotin/Repositories/Status/status-desktop/ui/StatusQ/include/StatusQ/keychain.h:9:7: note: previous definition is here
    9 | class LAContext;
      |       ^
```